server/src/main/java/org/apache/iotdb/db/mpp/sql/planner/DistributionPlanner.java

```diff
@@ -171,6 +175,29 @@ public PlanNode visitSchemaMerge(SchemaMergeNode node, DistributionPlanContext c
       return root;
     }
 
+    @Override
+    public PlanNode visitCountMerge(CountMergeNode node, DistributionPlanContext context) {
+      CountMergeNode root = (CountMergeNode) node.clone();
+      SchemaScanNode seed = (SchemaScanNode) node.getChildren().get(0);
+      Set<TRegionReplicaSet> schemaRegions = new HashSet<>();
+      analysis
+          .getSchemaPartitionInfo()
+          .getSchemaPartitionMap()
+          .forEach(
+              (storageGroup, deviceGroup) -> {
+                deviceGroup.forEach(
+                    (deviceGroupId, schemaRegionReplicaSet) ->
+                        schemaRegions.add(schemaRegionReplicaSet));
+              });
+      schemaRegions.forEach(
+          region -> {
+            SchemaScanNode schemaScanNode = (SchemaScanNode) seed.clone();
```

Comment:
We need to set a new PlanNodeId for newly generated SchemaScanNode using context

Reply:
Fixed